### PR TITLE
Allow hiding stderr during compilecache

### PR DESCRIPTION
**Pull request by IanButterworth**
Required for the error handling in https://github.com/JuliaLang/Pkg.jl/pull/2021

cc. @tkf

### CODE REVIEW

**Comment by tkf** [comment]
Context (just for record): Current implementation of `Pkg.precompile` as of https://github.com/JuliaLang/Pkg.jl/pull/2018 tries to call `Base.compilecache` for all packages in Manifest.toml file. This has a problem because some packages may only be required in some specific platform/environment and the package author advises the downstream packages to hide `using` behind, e.g., `if Sys.iswindows()` https://github.com/JuliaLang/Pkg.jl/pull/2018#discussion_r488853897. So, I suggested https://github.com/JuliaLang/Pkg.jl/pull/2018#discussion_r489029683 to ignore precompilation error from indirect dependencies (which is implemented in https://github.com/JuliaLang/Pkg.jl/pull/2021 by @ianshmean). Doing this in a user-friendly manner requires us to silence messages from the subprocesses.  Note that https://github.com/JuliaLang/Pkg.jl/pull/2021 does not ignore precompilation error from direct dependencies (which may be caused by precompilation of indirect dependencies). So, the errors relevant to the user will be correctly surfaced.

(It means that `Pkg.precompile` has quadratic time-complexity when an error happens. I think it's reasonable to optimize for the happy path for now given that doing this "correctly" is presumably much more challenging https://github.com/JuliaLang/Pkg.jl/pull/2018#discussion_r489029683. Also, we can still use `using TheProblematicPackage` to get precompilation error in linear time.)

File: base/loading.jl [hunk 1]
```diff
@@ -1210,7 +1210,7 @@ end
 
 const MAX_NUM_PRECOMPILE_FILES = 10
 
-function compilecache(pkg::PkgId, path::String)
+function compilecache(pkg::PkgId, path::String, show_errors::Bool = true)
```
Comment (tkf) [inline base/loading.jl:1213]: I think it'd be better to comment that `show_errors` is an "internal" interface for `Pkg.precompile`. It's a bit unclear why we have this from just looking at the code in Base.

**Comment by tkf** [comment]
@c42f I think you have opinions/thoughts on nested error message "handling" (in this case just ignoring). I think it's an OK implementation (though obviously not super elegant) but I wonder if you have some alternative ideas.

**Comment by c42f** [comment]
I haven't followed the larger story very closely but this seems fine to me, for now. Other options would probably be a lot more complicated.

**Comment by tkf** [comment]
> Other options would probably be a lot more complicated.

Yeah, that's my thought, too. Thanks for taking a look at it.

**Review by tkf — APPROVED** [review]
LGTM. I think it's pretty safe to merge but I guess I'll wait for a few days just in case someone (ping @vtjnash and @JeffBezanson) wants to take a look at it.

**Review by staticfloat — APPROVED** [review]
Looks good to me! Rebase and I say merge.

**Comment by ianshmean** [comment]
@staticfloat Rebased. Is the aarch64 issue relevant? `LoadError: LoadError: "hard kill repl test"`

**Pull request merged**